### PR TITLE
Review Simulation Control and General pages

### PR DESCRIPTION
**Pull request by jeannejoachim**
Make it more new-comer friendly: change the parameter order, reduce verbose and add missing parameters.

# Description of the problem

- Some formatting issues
- Difficult to read

# Description of the solution

- Fix formatting, add blocks when fitted
- Restructuring to increase readability for newcomers
- Reduce verbose
- Add some help messages there and there

# Documentation

- Parameters guide main page
- Simulation control

# Future changes

Feel free to add more content/explanations! If you like the spirit of these changes, I will review all the other parameter pages in the same fashion (for those which can still be improved).

### CODE REVIEW

File: doc/source/parameters/cfd/simulation_control.rst [hunk 1]
```diff
@@ -1,108 +1,177 @@
...
+.. seealso::
+	For further understanding about the numerical method used and advanced parameters, the interested reader is referred to the Theory Documentation.
 
 .. code-block:: text
 
-    # Adaptative time-stepping <true|false>
-    set adapt                        = false
+  subsection simulation control
+
+    # The kind of solver for the linear system
```
Comment (blaisb) [inline doc/source/parameters/cfd/simulation_control.rst:16]: This is not related to the linear system. It is the time stepping method.

File: doc/source/parameters/cfd/simulation_control.rst [hunk 2]
```diff
@@ -1,108 +1,177 @@
...
+    # BDF scheme parameters
+    #---------------------------------------------------
+    # Method used to startup high order BDF methods
+    set bdf startup method 	= multiple step bdf
+
+    # Scaling factor used in the iterations necessary to start-up the BDF schemes
+    set startup time scaling    = 0.4
+
+    #---------------------------------------------------
+    # Time-stepping parameters
```
Comment (blaisb) [inline doc/source/parameters/cfd/simulation_control.rst:38]: Transient simulations parameters

File: doc/source/parameters/cfd/simulation_control.rst [hunk 3]
```diff
@@ -1,108 +1,177 @@
...
-    # End time value of the simulation
-    set time end                     = 1
-  
-    # Time step value
-    set time step                    = 1.
+    set subdivision       = 1
+
   end
 
-* The ``group files`` parameter controls the number of vtu files generated in a parallel simulation. This parameter allows to reduce the number of files generated when the simulation is run with a large number of processors. Setting group files to 1 ensures that there is a single vtu file generated. In this case, the file is written using MPI IO functionalities. The value for this parameter should always be a compromise between keeping a low number of files but preventing excessive MPI communications. We have found that the default value of 1 does not have a significant impact on performance on Compute Canada clusters.
```
Comment (blaisb) [inline doc/source/parameters/cfd/simulation_control.rst:73]: I prefer the previous version. This is a very subtly parameter and I rather have it more verbose than not verbose

**Review by blaisb — CHANGES_REQUESTED** [review]
Initial review. I'll need to check it more

File: doc/source/parameters/cfd/simulation_control.rst [hunk 4]
```diff
@@ -1,108 +1,177 @@
...
+* ``method``: time-stepping method used. The available options are: 
+	* ``steady``: steady-state simulation
+	* ``steady-bdf``: steady-state simulation using adjoint time stepping with a bdf1 scheme
+	* ``bdf1``: 1st order backward differentiation
+	* ``bdf2``: 2nd order backward differentiation
+	* ``bdf3``: 3rd order backward differentiation
+	* ``sdirk2``: 2nd order singly diagonally implicit Runge Kutta
+	* ``sdirk3``: 3rd order singly diagonally implicit Runge Kutta
+
+* ``number of mesh adapt``: number of mesh adaptation during the steady-sate simulation
```
Comment (oguevremont) [inline doc/source/parameters/cfd/simulation_control.rst:102]:
```suggestion
* ``number mesh adapt``: number of mesh adaptations during the steady-sate simulation
```

File: doc/source/parameters/cfd/simulation_control.rst [hunk 5]
```diff
@@ -1,108 +1,177 @@
...
+    set startup time scaling    = 0.4
+
+    #---------------------------------------------------
+    # Time-stepping parameters
+    #---------------------------------------------------
+    # End time value of the simulation
+    set time end        = 1
+  
+    # Time step value
+    set time step       = 1.
```
Comment (oguevremont) [inline doc/source/parameters/cfd/simulation_control.rst:44]:
```suggestion
    set time step       = 1
```

Comment (oguevremont) [reply doc/source/parameters/cfd/simulation_control.rst:44]: The default is "1.", but to be homogeneous I'd remove the "."

File: doc/source/parameters/cfd/simulation_control.rst [hunk 6]
```diff
@@ -1,108 +1,177 @@
...
+	* ``steady-bdf``: steady-state simulation using adjoint time stepping with a bdf1 scheme
+	* ``bdf1``: 1st order backward differentiation
+	* ``bdf2``: 2nd order backward differentiation
+	* ``bdf3``: 3rd order backward differentiation
+	* ``sdirk2``: 2nd order singly diagonally implicit Runge Kutta
+	* ``sdirk3``: 3rd order singly diagonally implicit Runge Kutta
+
+* ``number of mesh adapt``: number of mesh adaptation during the steady-sate simulation
+
+* ``stop tolerance``: tolerance at which the adjoint time stepping steady state simulations (``method = steady-bdf``) stops.
```
Comment (oguevremont) [inline doc/source/parameters/cfd/simulation_control.rst:104]:
```suggestion
* ``stop tolerance``: tolerance at which the adjoint time stepping steady state simulation (``method = steady-bdf``) stops. 
```

File: doc/source/parameters/cfd/simulation_control.rst [hunk 7]
```diff
@@ -1,108 +1,177 @@
...
+    # Tolerance at which the simulation is stopped
+    set stop tolerance          = 1e-10
+
+    #---------------------------------------------------
+    # BDF scheme parameters
+    #---------------------------------------------------
+    # Method used to startup high order BDF methods
+    set bdf startup method 	= multiple step bdf
+
+    # Scaling factor used in the iterations necessary to start-up the BDF schemes
```
Comment (oguevremont) [inline doc/source/parameters/cfd/simulation_control.rst:34]:
```suggestion
    # Scaling factor used in the iterations necessary to startup the BDF schemes
```

File: doc/source/parameters/cfd/simulation_control.rst [hunk 8]
```diff
@@ -1,108 +1,177 @@
...
+
+.. note::
+	The adjoint time stepping will stop when the :math:`\mathcal{L}_2` norm of the initial residual is lower than ``stop tolerance`` at the start of a non-linear solution step.
+
+* ``bdf startup method``: scheme used to start a high order bdf scheme. The available options are: 
+	* ``multiple step bdf``
+	* ``sdirk step``
+	* ``initial solution``
+
+* ``startup time scaling``: scaling factor used in the iterations necessary to start-up the BDF schemes
```
Comment (oguevremont) [inline doc/source/parameters/cfd/simulation_control.rst:114]:
```suggestion
* ``startup time scaling``: scaling factor used in the iterations necessary to startup the BDF schemes
```

File: doc/source/parameters/cfd/simulation_control.rst [hunk 9]
```diff
@@ -1,108 +1,177 @@
...
+* ``output control``: control for the output of the simulation results. The available options are: 
+	* ``iteration``: results will be outputted at constant iteration frequency
+	* ``time`` : results will be outputted at constant time
+
+* ``output frequency``: controls after which number of iterations the ``.pvd`` / ``.vtu`` results are written. This parameter is only used when ``set output control = iteration``.
+
+.. tip::
+	If ``set output frequency = 0``, no ``.pvd`` / ``.vtu`` file will be written.
+
+	If the ``output frequency`` is set at a higher number than the total number of iterations in the simulation, the first (start-up) iteration will still be outputted.
```
Comment (oguevremont) [inline doc/source/parameters/cfd/simulation_control.rst:149]:
```suggestion
	If the ``output frequency`` is set at a higher number than the total number of iterations in the simulation, the first (startup) iteration will still be outputted.
```

File: doc/source/parameters/cfd/simulation_control.rst [hunk 10]
```diff
@@ -1,108 +1,177 @@
...
+	* ``sdirk3``: 3rd order singly diagonally implicit Runge Kutta
+
+* ``number of mesh adapt``: number of mesh adaptation during the steady-sate simulation
+
+* ``stop tolerance``: tolerance at which the adjoint time stepping steady state simulations (``method = steady-bdf``) stops. 
+
+.. note::
+	The adjoint time stepping will stop when the :math:`\mathcal{L}_2` norm of the initial residual is lower than ``stop tolerance`` at the start of a non-linear solution step.
+
+* ``bdf startup method``: scheme used to start a high order bdf scheme. The available options are:
```
Comment (oguevremont) [inline doc/source/parameters/cfd/simulation_control.rst:109]: Maybe we should specify if it's two and above?

File: doc/source/parameters/cfd/simulation_control.rst [hunk 11]
```diff
@@ -1,108 +1,177 @@
...
+
+.. tip::
+	If ``set output frequency = 0``, no ``.pvd`` / ``.vtu`` file will be written.
+
+	If the ``output frequency`` is set at a higher number than the total number of iterations in the simulation, the first (start-up) iteration will still be outputted.
+
+
+* ``output time``: controls the time when the ``.pvd`` / ``.vtu`` results are written. This parameter is only used when ``set output control = time``.
+
+* ``group files``: number of ``vtu`` files generated in a parallel simulation
```
Comment (oguevremont) [inline doc/source/parameters/cfd/simulation_control.rst:154]:
```suggestion
* ``group files``: number of ``.vtu`` files generated in a parallel simulation
```

**Review by oguevremont — COMMENTED** [review]
Nice work! I like the format. I suggested some changes to be more homogeneous, for example with startup

File: doc/source/parameters/cfd/simulation_control.rst (hunk 3 above)
Comment (jeannejoachim) [reply doc/source/parameters/cfd/simulation_control.rst:73]: Don't worry, I kept all this information, this is of course relevant! I just put it in the tip-box just below, I think it is clearer when displayed in a web browser

File: doc/source/parameters/cfd/simulation_control.rst (hunk 4 above)
Comment (jeannejoachim) [reply doc/source/parameters/cfd/simulation_control.rst:102]: eyes of the tiger! :tiger: ;)

File: doc/source/parameters/cfd/simulation_control.rst [hunk 12]
```diff
@@ -1,108 +1,170 @@
...
 
-* The ``time end`` parameter indicates at which value of the time to end the simulations if they are transient.
+* ``group files``: number of ``.vtu`` files generated in a parallel simulation
 
-* The ``time step`` parameter controls the value of the time step.
+.. tip::
+	This parameter allows to reduce the number of files generated when the simulation is run with a large number of processors. ``set group files = 1`` ensures that a single ``.vtu`` file will be generated. In this case, the file is written using MPI IO functionalities.
 
-* The ``adapt`` parameter controls if adaptive time-stepping is enabled. If adaptive time-stepping is enabled, the time-step will evolve to ensure that the 'max cfl' value is reached.
+	The value for this parameter should always be a compromise between keeping a low number of files but preventing excessive MPI communications. We have found that the default value of 1 does not have a significant impact on performance on Compute Canada clusters.
```
Comment (blaisb) [inline doc/source/parameters/cfd/simulation_control.rst:159]:
```suggestion
	The value for this parameter should always be a compromise between keeping a low number of files but preventing excessive MPI communications. We have found that the default value of 1 does not have a significant impact on performance on Compute Canada clusters. However, as soon as the size of the output vtu file reaches 1Gb, it is preferable  to start splitting them into multiple smaller files as this may lead to corrupted files on some file systems.
```

**Review by blaisb — APPROVED** [review]
Just one last comment.

File: doc/source/parameters/cfd/simulation_control.rst (hunk 12 above)
Comment (jeannejoachim) [reply doc/source/parameters/cfd/simulation_control.rst:159]: relevant indeed! Thank you
